### PR TITLE
Feature/163/user security

### PR DESCRIPTION
**Pull request by GrazhdanochkaMariya**
Added uuid to user. Now we can see the details and edit with url+uuid

### CODE REVIEW

**Comment by gluhovSemen** [comment]
> Added uuid to user. Now we can see the details and edit with url+uuid

Why do we need uuid in url?

File: src/apps/user/views.py [hunk 1]
```diff
@@ -1,29 +1,38 @@
 from django.contrib.auth import get_user_model
 from django.contrib.auth.mixins import LoginRequiredMixin
-from django.urls import reverse_lazy
+from django.shortcuts import get_object_or_404
+from django.urls import reverse_lazy, reverse
 from django.views.generic import UpdateView, DetailView
 
 from src.apps.user.forms import CustomUserChangeForm, ProfileForm
-from src.apps.user.models import UserProfile, UserAddress
+from src.apps.user.models import UserProfile, UserAddress, User
```
Comment (egorvavilov) [inline src/apps/user/views.py]: Dont import User model like this

use "User = get_user_model()" instead

File: src/apps/user/views.py [hunk 2]
```diff
@@ -1,29 +1,38 @@
...
 
 
 class UserDetailView(LoginRequiredMixin, DetailView):
     template_name = "user/user_detail.html"
     form_class = CustomUserChangeForm
-    success_url = reverse_lazy("user_detail")
 
     def get_object(self, queryset=None):
-        return self.request.user
+        uuid = self.kwargs.get("uuid")
```
Comment (egorvavilov) [inline src/apps/user/views.py]: Take a look at "class SingleObjectMixin(ContextMixin)"
I think we can find better solution to that)

Hint: review slug;)

File: src/apps/user/views.py [hunk 3]
```diff
@@ -1,29 +1,38 @@
...
     template_name = "user/user_edit.html"
-    success_url = reverse_lazy("user_detail")
 
     def get_object(self, queryset=None):
-        return self.request.user
+        uuid = self.kwargs.get("uuid")
+        return get_object_or_404(User, uuid=uuid)
+
+    def get_success_url(self):
+        uuid = self.kwargs.get("uuid")
```
Comment (egorvavilov) [inline src/apps/user/views.py]: take a look at "class RedirectURLMixin:"
Can we use default class methods?

**Review by egorvavilov — CHANGES_REQUESTED** [review]

**Comment by GrazhdanochkaMariya** [comment]
> > Added uuid to user. Now we can see the details and edit with url+uuid
> 
> Why do we need uuid in url?

We need it it for the future features like most_viewed_products, last_user_viewed_products, when we should get the unique user for this goal. This is  the new business requirement. :)

File: src/apps/user/views.py (hunk 3 above)
Comment (GrazhdanochkaMariya) [reply src/apps/user/views.py]: I believe we can. Will take a look!

**Review by egorvavilov — APPROVED** [review]

**Pull request merged**